packages/ses/src/error/unhandled-rejection.js

```diff
@@ -0,0 +1,121 @@
+// @ts-check
+import {
+  FinalizationRegistry,
+  Map,
+  mapGet,
+  mapDelete,
+  WeakMap,
+  mapSet,
+  finalizationRegistryRegister,
+  weakmapSet,
+  weakmapGet,
+  mapEntries,
+} from '../commons.js';
+
+/**
+ * Create rejection-tracking machinery compatible with Node.js and browsers.
+ *
+ * Note that modern browsers *prevent* access to the 'unhandledrejection' and
+ * 'rejectionhandled' events needed:
+ * - in cross-origin mode, like when served from file://
+ * - in the browser console (interactively typed-in code)
+ * - in the debugger
+ *
+ * Then, they just look like: `Uncaught (in promise) Error: ...` and don't
+ * implement the machinery.
+ *
+ * The solution is to serve your web page from an http:// or https:// web server
+ * and execute actual code.
+ *
+ * @param {(reason: unknown) => void} reportReason report the reason for an
+ * unhandled rejection.
+ */
+export const makeRejectionHandlers = reportReason => {
+  if (FinalizationRegistry === undefined) {
+    return undefined;
+  }
+
+  /** @typedef {number} ReasonId */
+  let lastReasonId = 0;
+
+  /** @type {Map<ReasonId, unknown>} */
+  const idToReason = new Map();
+
+  /** @type {(() => void) | undefined} */
+  let cancelChecking;
+
+  const removeReasonId = reasonId => {
+    mapDelete(idToReason, reasonId);
+    if (cancelChecking && idToReason.size === 0) {
+      // No more unhandled rejections to check, just cancel the check.
+      cancelChecking();
+      cancelChecking = undefined;
+    }
+  };
+
+  /** @type {WeakMap<Promise, ReasonId>} */
+  const promiseToReasonId = new WeakMap();
+
+  /**
+   * Clean up and report the reason for a GCed unhandled rejection.
+   *
+   * @param {ReasonId} heldReasonId
+   */
+  const finalizeDroppedPromise = heldReasonId => {
+    const reason = mapGet(idToReason, heldReasonId);
+    if (reason !== undefined) {
```

Comment:
Nit, technically you can reject with `undefined` value...
```suggestion
    if (mapHas(idToReason, heldReasonId)) {
      const reason = mapGet(idToReason, heldReasonId);
```